video/out/w32_common.c

```diff
@@ -766,35 +766,80 @@ static DWORD update_style(struct vo_w32_state *w32, DWORD style)
     return style;
 }
 
-// Update the window title, position, size, and border style.
-static void reinit_window_state(struct vo_w32_state *w32)
+// Adjust window size if it is larger than the virtual screen size.
+static void fit_window_on_screen(struct vo_w32_state *w32, RECT *rc)
 {
-    HWND layer = HWND_NOTOPMOST;
-    RECT r;
+    if (w32->current_fs || IsMaximized(w32->window))
+        return;
 
-    if (w32->parent)
+    RECT b = {0};
+    add_window_borders(w32->window, &b);
+
+    // Check on current window size, add borders on --fit-border
+    int o_w = rc->right - rc->left;
+    int o_h = rc->bottom - rc->top;
+    if (w32->opts->fit_border) {
+        o_w += b.right - b.left;
+        o_h += b.bottom - b.top;
+    }
+
+    MP_VERBOSE(w32, "requested window size: %d:%d\n", o_w, o_h);
+
+    // Use virtual screen size, since the mpv window can be resized
+    // to cover mupltiple monitors
+    const int screen_w = GetSystemMetrics(SM_CXVIRTUALSCREEN);
+    const int screen_h = GetSystemMetrics(SM_CYVIRTUALSCREEN);
+
+    if (o_w <= screen_w && o_h <= screen_h) {
+        MP_VERBOSE(w32, "window already fits on the screen\n");
         return;
+    }
 
-    bool new_fs = w32->toggle_fs ? !w32->current_fs : w32->opts->fullscreen;
-    bool toggle_fs = w32->current_fs != new_fs;
-    w32->current_fs = new_fs;
-    w32->toggle_fs = false;
+    MP_VERBOSE(w32, "adjusting window size to fit it on the screen\n");
```

Comment:
This is a bit too chatty. You can tell if the window size was adjusted from the "adjusted window size" message at the end of the function. Verbose-mode is very noisy as-is.